internal/tools/aws/filestore.go

```diff
@@ -35,19 +37,39 @@ func (f *S3Filestore) Provision(logger log.FieldLogger) error {
 
 // Teardown removes all AWS resources related to an S3 filestore.
 func (f *S3Filestore) Teardown(keepData bool, logger log.FieldLogger) error {
-	err := s3FilestoreTeardown(f.installationID, keepData, logger)
+	logger.Info("Tearing down AWS S3 filestore")
+
+	awsID := CloudID(f.installationID)
+
+	err := f.awsClient.iamEnsureUserDeleted(awsID, logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to teardown AWS S3 filestore")
+	}
+
+	err = f.awsClient.secretsManagerEnsureIAMAccessKeySecretDeleted(awsID, logger)
 	if err != nil {
 		return errors.Wrap(err, "unable to teardown AWS S3 filestore")
 	}
 
+	if keepData {
+		logger.WithField("s3-bucket-name", awsID).Info("AWS S3 bucket was left intact due to the keep-data setting of this server")
+		return nil
+	}
+
+	err = f.awsClient.s3EnsureBucketDeleted(awsID, logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to teardown AWS S3 filestore")
```

Comment:
can we change the language in this error to "unable to ensure AWS S3 filestore was deleted" or something similar, just so that it's not identical to the error on line 46?

Reply:
Same as above, code was already there but doesn't hurt to improve it.